### ISSUE TITLE
Rule L030 consistency expectation doesn't seem to exclude templated code.

### ISSUE THREAD

**Issue by cwcartmell** [sqlfluff/sqlfluff#2565]
I believe that using templated code that generates a function before using an un-templated function in a query can cause an inconsistent capitalization error. If you reverse the order, this seems to be not true.

## Expected Behaviour

Both of the following models:

```sql
-- test_model_1.sql
SELECT
    1 AS i,
    2 AS j,
    {{ dbt_utils.surrogate_key(["i", "j"]) }} AS ij_id,
    GREATEST(i, j) AS max_ij
```

```sql
-- test_model_2.sql
SELECT
    1 AS i,
    2 AS j,
    {{ dbt_utils.surrogate_key(["i", "j"]) }} AS ij_id,
    GREATEST(i, j) AS max_ij
```

Lint without raising an error for rule L030 given the configuration specified below.

## Observed Behaviour

```zsh
$ sqlfluff lint
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [/Users/conradcartmell/Documents/sqlfluff_testing/models/test_model_1.sql] FAIL
L:   5 | P:   5 | L030 | Function names must be consistently lower case.
All Finished 📜 🎉!
```

I believe this error is being raised because the dbt_utils `surrogate_key` function generates SQL with lowercase functions, and although by default `ignore_templated_areas` is set to `True` SQLFluff is still basing its expectations for the capitalization of subsequent SQL functions on those templated functions.

## Steps to Reproduce

* Create a dbt project with the following packages.yml file:

```yml
packages:
- package: dbt-labs/dbt_utils
  version: 0.7.6

```

* Add the models in "observed behavior" to the dbt project.
* Place the included pyproject.toml file in the dbt project directory.
* Run `sqlfluff lint` in the dbt project directory.

## Dialect

ANSI

## Version

```zsh
$ sqlfluff version
0.9.4
$ python --version
3.8.8
```

## Configuration

pyproject.toml:
```toml
[tool.sqlfluff.core]
templater = "dbt"

[tool.sqlfluff.templater.dbt]
project_dir = "."

# SQL function names must be either consistently UPPER-CASE or consistently lower-case.
[tool.sqlfluff.rules.L030]
capitalisation_policy = "consistent"
```
 Raw config:
```
core:
    dialect:            ansi
    disable_noqa:       False
    encoding:           autodetect
    ignore:             []
    ignore_templated_areas:True
    nocolor:            False
    output_line_length: 80
    recurse:            True
    runaway_limit:      10
    sql_file_exts:      .sql,.sql.j2,.dml,.ddl
    templater:          dbt
    verbose:            2
indentation:
    indented_joins:     False
    indented_using_on:  True
    template_blocks_indent:True
rules:
    allow_scalar:       True
    comma_style:        trailing
    indent_unit:        space
    max_line_length:    80
    single_table_references:consistent
    tab_space_size:     4
    unquoted_identifiers_policy:all
    L007:
        operator_new_lines: after
    L010:
        capitalisation_policy:consistent
    L011:
        aliasing:           explicit
    L012:
        aliasing:           explicit
    L014:
        extended_capitalisation_policy:consistent
    L016:
        ignore_comment_lines:False
    L026:
        force_enable:       False
    L028:
        force_enable:       False
    L029:
        unquoted_identifiers_policy:aliases
    L030:
        capitalisation_policy:consistent
    L038:
        select_clause_trailing_comma:forbid
    L040:
        capitalisation_policy:consistent
    L042:
        forbid_subquery_in: join
    L047:
        prefer_count_0:     False
        prefer_count_1:     False
    L052:
        multiline_newline:  False
        require_final_semicolon:False
    L054:
        group_by_and_order_by_style:consistent
    L057:
        additional_allowed_characters:""
        allow_space_in_identifier:False
        quoted_identifiers_policy:all
        unquoted_identifiers_policy:all
templater:
    unwrap_wrapped_queries:True
    dbt:
        project_dir:        .
    jinja:
        apply_dbt_builtins: True
        macros:
            dbt_config:         {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
            dbt_is_incremental: {% macro is_incremental() %}True{% endmacro %}
            dbt_ref:            {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
            dbt_source:         {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
            dbt_var:            {% macro var(variable, default='') %}item{% endmacro %}
```

**Comment by barrywhart** [comment]
Ah yes, I can see that. We have been able to fix lots of "templated code" issues in the core linter, but in this case, the rule itself needs to be updated to ignore templated code.

Note: The function`Segments.raw_slices()` will be very helpful for this.
